app/models/event.rb

```diff
@@ -88,9 +88,12 @@ def to_s
   end
 
   def members
-    @members or @members = event_roles.inject(Array.new) do |uniq_roles, er| 
-      ( uniq_roles << er.member unless er.member.nil? or uniq_roles.any? { |ur| ur.id == er.member_id } ) or uniq_roles 
+    if @members
+      return @members
     end
+
+    non_unique_members = event_roles.map(&:member).select{ |m| not m.nil?}
+    @members = non_unique_members.uniq(&:m.id)
```

Comment:
I don't think `m` is defined outside of the `select` in the above line. Should this be `uniq(&:id)` or `uniq(&:member.id)`? I am not certain. Has this been tested?

We should make sure this returns an empty array not not `nil` if there are no results in the `map`/`select` since the previous `inject(Array.new)` ensured that.

This has to be a speed improvement.

Reply:
Yeah, the code is wrong. However, looking back I can't find any uses of this function or the members attribute.